api/client.go

```diff
@@ -1,26 +1,101 @@
 package api
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
-	url "net/url"
+	"net/url"
 
+	skyapi "gitlab.com/SkynetLabs/skyd/node/api"
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+
+	"github.com/SkynetLabs/blocker/database"
 	"gitlab.com/NebulousLabs/errors"
 	"gitlab.com/SkynetLabs/skyd/node/api"
 )
 
-// Client is a helper struct that gets initialised using a portal url. It
-// exposes API methods and abstracts the response handling.
-type Client struct {
-	staticPortalURL string
-}
+const (
+	// clientDefaultTimeout is the timeout of the http calls to in seconds
+	clientDefaultTimeout = "30"
+)
+
+type (
+	// Client is a helper struct that gets initialised using a portal url. It
+	// exposes API methods and abstracts the response handling.
+	Client struct {
```

Comment:
so the whole was for it to be a generic client actually, but I think I should wrap it in a `SkydClient` struct and return that from `NewSkydClient`. I just want to stop re-defining `.get` and `.post` for generic APIs. i will do this in a F/U but I'll tackle it almost immediately, so it will get done 👍